File: loom-compiler-old/src/main/java/loom/compiler/ast/Parameter.java
package loom.compiler.ast;

import loom.compiler.token.TokenSpan;

public class Parameter {
	public final String name;
	public final TypeNode type;
	public final TokenSpan span;

	public Parameter(String name, TypeNode type, TokenSpan span) {
		this.name = name;
		this.type = type;
		this.span = span;
	}

	@Override
	public String toString() {
		return "Parameter{name='" + name + "', type=" + type + '}';
	}
}

File: Frontend/Parser/AST/Expressions/Parameter.cs
// Copyright (c) 2026.The Cloth contributors.
// 
// Parameter.cs is part of the Cloth Frontend.
// 
// Use, modification, and distribution of this file are governed by the
// license terms provided with the Cloth Compiler source distribution.

using FrontEnd.Parser.AST.Type;
using FrontEnd.Token;

namespace FrontEnd.Parser.AST.Expressions;

public readonly record struct Parameter(TypeExpression Type, string Name, Expression? Default, TokenSpan Span);